src/components/UserProfile/UserProfileSettings/index.js

```diff
@@ -8,14 +8,24 @@ import { connect } from "react-redux";
 
 import StyledButton from "../../common/StyledButton";
 import { useStyles } from "./styles";
-import { userActions } from "../../../Redux/actionCreators";
+import { userActions, loaderActions } from "../../../Redux/actionCreators";
 import Routes from "../../../utility/constants/Routes";
-import AlertBox from "../../common/AlertBox";
+import AlertBox, { alertTypes } from "../../common/AlertBox";
+import ConfirmDelete from "./ConfirmDelete";
 
 class UserProfileSettings extends Component {
   state = {
-    error: undefined,
-    emailAlerts: this.props.emailAlerts,
+    alertMessage: undefined,
+    alertType: alertTypes.ERROR,
+    emailAlerts: false,
+    showConfirmDelete: false,
+    confirmDeleteError: undefined,
+  };
+
+  componentDidUpdate = prevProps => {
+    if (prevProps.emailAlerts !== this.props.emailAlerts) {
+      this.setState({ emailAlerts: this.props.emailAlerts });
```

Comment:
Can we not directly use `props.emailAlerts` instead of storing it in state. Storing props in state is not needed in most of the cases. You can find a very good writeup about derived state [here](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html). As it mentions use derived state sparingly.

Reply:
It is reqd because.. The local state has to be synced with the props initially and later, the state will be edited independently.. Once the change is submitted over API, it has to be synced again.

Reply:
Is it possible to move that sate to redux completely? We are anyway partially relying on redux for this.

Reply:
https://github.com/singnet/refactored-dapp/pull/116/commits/f5c8fa383acf9570cf9da77f0e88138ac8f065d7